File: cookiecutter-dotnet-template/{{cookiecutter.project_slug}}/Controllers/HelloController.cs
using Microsoft.AspNetCore.Mvc;
using {{cookiecutter.namespace}}.Domain.Models;
using {{cookiecutter.namespace}}.Domain.Services;

namespace {{cookiecutter.namespace}}.Controllers
{
    [ApiController]
    [Route("hello")]
    public class HelloController : ControllerBase
    {
        private readonly IGreetingService _greetingService;

        public HelloController(IGreetingService greetingService)
        {
            _greetingService = greetingService;
        }

        [HttpGet("{name?}")]
        public ActionResult<Greeting> Get(string? name)
        {
            return _greetingService.Greet(name);
        }
    }
}

File: cookiecutter-spring-mvn-template/{{cookiecutter.project_slug}}/src/main/java/{{cookiecutter.package_path}}/infrastructure/web/HelloController.java
package {{cookiecutter.package_name}}.infrastructure.web;

import {{cookiecutter.package_name}}.domain.model.Greeting;
import {{cookiecutter.package_name}}.domain.service.GreetingService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/hello")
public class HelloController {

    private final GreetingService greetingService;

    public HelloController(GreetingService greetingService) {
        this.greetingService = greetingService;
    }

    @GetMapping("/{name}")
    public Greeting hello(@PathVariable String name) {
        return greetingService.greet(name);
    }
}
